### PR TITLE
slack notifications support added

### PR DESCRIPTION
**Pull request by arturdobo**
I added support for slack notifications

### CODE REVIEW

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class SlackNotifierTest {
+	private final String channel = "channel";
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java]: Codacy Issue found: [This final field could be made static](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2396352067/issues/source?bid=3175464&fileBranchId=3231081#l25)

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	private final String icon = "icon";
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java]: Codacy Issue found: [This final field could be made static](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2396352067/issues/source?bid=3175464&fileBranchId=3231081#l26)

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	private final String user = "user";
+	private final String appName = "App";
+	private final String id = "-id-";
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java]: Codacy Issue found: [This final field could be made static](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2396352067/issues/source?bid=3175464&fileBranchId=3231081#l29)

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+				           .withHealthUrl("http://health")
+				           .build(),
+				infoDown, infoUp);
+	}
+
+	private Object expectedMessage(String color, String user, @Nullable String icon,
+	                               @Nullable String channel, String message) {
+		Map<String, Object> messageJson = new HashMap<>();
+		messageJson.put("username", user);
+		if (icon != null) messageJson.put("icon_emoji", ":" + icon + ":");
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java]: Codacy Issue found: [Avoid using if statements without curly braces](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2396352067/issues/source?bid=3175464&fileBranchId=3231081#l131)

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class SlackNotifierTest {
+	private final String channel = "channel";
+	private final String icon = "icon";
+	private final String user = "user";
+	private final String appName = "App";
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java]: Codacy Issue found: [This final field could be made static](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2396352067/issues/source?bid=3175464&fileBranchId=3231081#l28)

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+				           .build(),
+				infoDown, infoUp);
+	}
+
+	private Object expectedMessage(String color, String user, @Nullable String icon,
+	                               @Nullable String channel, String message) {
+		Map<String, Object> messageJson = new HashMap<>();
+		messageJson.put("username", user);
+		if (icon != null) messageJson.put("icon_emoji", ":" + icon + ":");
+		if (channel != null) messageJson.put("channel", channel);
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java]: Codacy Issue found: [Avoid using if statements without curly braces](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2396352067/issues/source?bid=3175464&fileBranchId=3231081#l132)

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class SlackNotifierTest {
+	private final String channel = "channel";
+	private final String icon = "icon";
+	private final String user = "user";
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java]: Codacy Issue found: [This final field could be made static](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2396352067/issues/source?bid=3175464&fileBranchId=3231081#l27)

File: spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+	private final String appName = "App";
+	private final String id = "-id-";
+	private final String message = "test";
```
Comment (joshiste) [inline spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/SlackNotifierTest.java]: Codacy Issue found: [This final field could be made static](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2396352067/issues/source?bid=3175464&fileBranchId=3231081#l30)

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/notify/SlackNotifier.java [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+	public void setMessage(Expression message) {
+		this.message = message;
+	}
+
+	private Object createMessage(ClientApplicationStatusChangedEvent event) {
+		Map<String, Object> messageJson = new HashMap<>();
+		messageJson.put("username", username);
+		if (icon != null) messageJson.put("icon_emoji", ":" + icon + ":");
+		if (channel != null) messageJson.put("channel", channel);
```
Comment (joshiste) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/notify/SlackNotifier.java]: Codacy Issue found: [Avoid using if statements without curly braces](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2396352040/issues/source?bid=3175464&fileBranchId=3231081#l87)

File: spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/notify/SlackNotifier.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+	}
+
+	public void setMessage(Expression message) {
+		this.message = message;
+	}
+
+	private Object createMessage(ClientApplicationStatusChangedEvent event) {
+		Map<String, Object> messageJson = new HashMap<>();
+		messageJson.put("username", username);
+		if (icon != null) messageJson.put("icon_emoji", ":" + icon + ":");
```
Comment (joshiste) [inline spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/notify/SlackNotifier.java]: Codacy Issue found: [Avoid using if statements without curly braces](https://www.codacy.com/app/joshiste/spring-boot-admin/file/2396352040/issues/source?bid=3175464&fileBranchId=3231081#l86)

**Comment by joshiste** [comment]
Hi arturdobo thank you for your contribution.
Can you please fix the issues found by codacy and use the provided [eclipse formatter](https://github.com/codecentric/spring-boot-admin/blob/master/eclipse-code-formatter.xml)?
In case you are sing intelliJ the [eclipse code formatter plugin](https://plugins.jetbrains.com/plugin/6546) will help you.

**Comment by arturdobo** [comment]
Hi, Sorry, I didn't notice eclipse formatter earlier. 
I fixed all issues and now I think it looks properly.

**Comment by joshiste** [comment]
Merged with bd48357ec3372170b4431443808e4ca9cc5a8e99
Thanks for contributing!

**Pull request closed without merging**